src/Control/Carrier/NonDet/Church.hs

```diff
@@ -75,14 +77,17 @@ newtype NonDetC m a = NonDetC (forall b . (m b -> m b -> m b) -> (a -> m b) -> m
 instance Applicative (NonDetC m) where
   pure a = NonDetC (\ _ leaf _ -> leaf a)
   {-# INLINE pure #-}
+
   NonDetC f <*> NonDetC a = NonDetC $ \ fork leaf nil ->
     f fork (\ f' -> a fork (leaf . f') nil) nil
   {-# INLINE (<*>) #-}
 
 instance Alternative (NonDetC m) where
   empty = NonDetC (\ _ _ nil -> nil)
   {-# INLINE empty #-}
-  NonDetC l <|> NonDetC r = NonDetC $ \ fork leaf nil -> fork (l fork leaf nil) (r fork leaf nil)
+
+  NonDetC l <|> NonDetC r = NonDetC $ \ fork leaf nil ->
+    l fork leaf nil `fork` r fork leaf nil
```

Comment:
I’ve also taken the opportunity to write `fork` infix in this module and the `ChooseC` one to underscore its purpose.